.gitignore

```diff
@@ -133,3 +133,6 @@ dmypy.json
 methods/transient/calval
 methods/transient/.ipynb_checkpoints/
 methods/transient/error.log
+methods/transient/NISAR_SES_Secular_Requirement_Validation_v2.ipynb
+methods/transient/calval_secular/
+methods/transient/DataHoldings.txt
```

Comment:
My guess is that you are running the notebook in the same directory as your data files. Notebooks are more of code, I would recommend you to separate them into your different local directories. Then all these "methods" types are not needed anymore in the `.gitignore` file.

Reply:
I found when I call the `mintpy.objects.gps.search_gps`, a `DataHoldings.txt' will be written into the current directory. I can use `os.chdir` to change the current directory to the data processing directory when I call this function.

I use `calval_secular` and ` NISAR_SES_Secular_Requirement_Validation_v2.ipynb` to check if the transient validation approach notebook 1 can also be used for secular requirement. These two should not be included in the transient method directory. I will move them out.

Reply:
You could use `gps.search_gps(site_list_file)` argument to specify the text file location, to overwrite the default current directory, then no `os.chdir()` is needed.

Reply:
@yunjunz I find a potential typo in `gps.search_gps()`:
```
def search_gps(SNWE, start_date=None, end_date=None, site_list_file=None, min_num_solution=50, [print_msg=True):
    ...
    # download site list file if it's not found in current directory
    if site_list_file is None:
        site_list_file = os.path.basename(unr_site_list_file)
        if not os.path.isfile(site_list_file):
            dload_site_list(print_msg=print_msg)
```
The `site_list_file` will not be downloaded if it is specified by the user. If we change the code to
```
    if site_list_file is None:
        site_list_file = os.path.basename(unr_site_list_file)
    if not os.path.isfile(site_list_file):
        dload_site_list(print_msg=print_msg)
```
It will work.

Reply:
Yes, that's a bug, could you issue a PR to the mintpy repo to fix it?

Reply:
Sure. I am happy to fix it.